File: sasmodels/models/sasfit_disc_sd_bilayergauss.c
///////////////////////////////////////////////////
//    This is automatically genearted file       //
//    by sasfit_convert.py                       //
//    Some editting might be required            //
///////////////////////////////////////////////////

double Iq( double q, double T,  double SIGMA_T,  double D,  double SIGMA_D,
           double DUMMY,  double DUMMY,  double SIGMA_OUT,  double B_OUT,
           double SIGMA_CORE,  double B_CORE,  double P0,  double LB,  double B,  double L,
           double EXVOL,  double CLB);
double Fq( double q,  double T,  double SIGMA_T,  double D,  double SIGMA_D,
           double DUMMY,  double DUMMY,  double SIGMA_OUT,  double B_OUT,
           double SIGMA_CORE,  double B_CORE,  double P0,  double LB,  double B,  double L,
           double EXVOL,  double CLB);
double form_volume(  double T,  double SIGMA_T,  double D,  double SIGMA_D,
                     double DUMMY,  double DUMMY,  double SIGMA_OUT,  double B_OUT,
                     double SIGMA_CORE,  double B_CORE,  double P0,  double LB,  double B,  double L,
                     double EXVOL,  double CLB);
double Iqxy( double qx, double qy, double T, double SIGMA_T, double D,
             double SIGMA_D, double DUMMY, double DUMMY, double SIGMA_OUT, double B_OUT,
             double SIGMA_CORE, double B_CORE, double P0, double LB, double B, double L,
             double EXVOL, double CLB);
/*
* Author(s) of this file:
*   Joachim Kohlbrecher (joachim.kohlbrecher@psi.ch)
*/
// define shortcuts for local T, SIGMA_T, D, SIGMA_D, DUMMY, DUMMY, SIGMA_OUT, B_OUT, SIGMA_CORE, B_CORE, P0, LB, B, L, EXVOL, CLBeters/variables
double Iq( double q, double T,  double SIGMA_T,  double D,  double SIGMA_D,
           double DUMMY,  double DUMMY,  double SIGMA_OUT,  double B_OUT,
           double SIGMA_CORE,  double B_CORE,  double P0,  double LB,  double B,  double L,
           double EXVOL,  double CLB)
{
// insert your code here
    return 0.0;
}
double Fq( double q,  double T,  double SIGMA_T,  double D,  double SIGMA_D,
           double DUMMY,  double DUMMY,  double SIGMA_OUT,  double B_OUT,
           double SIGMA_CORE,  double B_CORE,  double P0,  double LB,  double B,  double L,
           double EXVOL,  double CLB)
{
// insert your code here
    return 0.0;
}
double form_volume(  double T,  double SIGMA_T,  double D,  double SIGMA_D,
                     double DUMMY,  double DUMMY,  double SIGMA_OUT,  double B_OUT,
                     double SIGMA_CORE,  double B_CORE,  double P0,  double LB,  double B,  double L,
                     double EXVOL,  double CLB)
{
// insert your code here
    return 0.0;
}
double Iqxy( double qx, double qy, double T, double SIGMA_T, double D,
             double SIGMA_D, double DUMMY, double DUMMY, double SIGMA_OUT, double B_OUT,
             double SIGMA_CORE, double B_CORE, double P0, double LB, double B, double L,
             double EXVOL, double CLB)
{
    double q = sqrt(qx*qx + qy*qy);
    return Iq( q, T, SIGMA_T, D, SIGMA_D, DUMMY, DUMMY, SIGMA_OUT, B_OUT,
               SIGMA_CORE, B_CORE, P0, LB, B, L, EXVOL, CLB);
}

File: sasmodels/models/sasfit_disc_sd_bilayergauss.py
r"""
This file has been automatically gereated by sasfit_convert
The model calculates an empirical functional form for SAS data characterized
by disc_sd_bilayergauss

Definition:
-----------

References:
-----------

"""
from numpy import inf

name = "disc_sd_bilayergauss"
title = " "
description = ""
category = " "
#pylint: disable=bad-whitespace, line-too-long
parameters = [
 [ "T", 	"", 	10.0, 	[-inf, inf], 	"", 	""],
 [ "SIGMA_T", 	"", 	0.0, 	[-inf, inf], 	"", 	""],
 [ "D", 	"", 	0.0, 	[-inf, inf], 	"", 	""],
 [ "SIGMA_D", 	"", 	1.0, 	[-inf, inf], 	"", 	""],
 [ "DUMMY", 	"", 	0.0, 	[-inf, inf], 	"", 	""],
 [ "DUMMY", 	"", 	0.0, 	[-inf, inf], 	"", 	""],
 [ "SIGMA_OUT", 	"", 	0.0, 	[-inf, inf], 	"", 	""],
 [ "B_OUT", 	"", 	0.0, 	[-inf, inf], 	"", 	""],
 [ "SIGMA_CORE", 	"", 	0.0, 	[-inf, inf], 	"", 	""],
 [ "B_CORE", 	"", 	0.0, 	[-inf, inf], 	"", 	""],
 [ "P0", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "LB", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "B", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "L", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "EXVOL", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
 [ "CLB", 	"", 	0.0, 	[-inf, inf], 	"", 	" "],
]
 #pylint: enable=bad-whitespace, line-too-long

source = [  "sasfit_disc_sd_bilayergauss.c" ]

demo = dict(
	T = 10.0,
	SIGMA_T = 0.0,
	D = 0.0,
	SIGMA_D = 1.0,
	DUMMY = 0.0,
	DUMMY = 0.0,
	SIGMA_OUT = 0.0,
	B_OUT = 0.0,
	SIGMA_CORE = 0.0,
	B_CORE = 0.0,
	P0 = 0.0,
	LB = 0.0,
	B = 0.0,
	L = 0.0,
	EXVOL = 0.0,
	CLB = 0.0)
